scipy/sparse/linalg/tests/test_propack.py

```diff
@@ -76,46 +80,89 @@ def test_svdp(ctor, dtype, irl, which):
         check_svdp(n, m, ctor, dtype, k, irl, which)
 
 
-def load_coord(folder, precision, file="illc1850.coord"):
-    dtype = {"single": np.float32, "double": np.float64}[precision]
-    path = os.path.join(folder, precision, file)
-    with open(path) as f:
-        m, n, nnz = (int(val) for val in f.readline().split())
-        coord = np.array([[float(val) for val in line.split()] for line in f])
-    i = coord[:, 0].astype(int) - 1
-    j = coord[:, 1].astype(int) - 1
-    data = coord[:, 2].astype(dtype)
-    A = coo_matrix((data, (i, j)))
+def load_coord(folder, precision, file=None):
+    dtype = {
+        "single": np.float32,
+        "double": np.float64,
+        "complex8": np.complex64,
+        "complex16": np.complex128,
+    }[precision]
+    if file is None:
+        file = "illc1850.coord" if precision in {'single', 'double'} else 'mhd1280b.cua'
+    path = os.path.join(folder, precision, 'Examples', file)
+    if precision in {'single', 'double'}:
+        # Coordinate Text File
+        with open(path) as f:
+            m, n, nnz = (int(val) for val in f.readline().split())
+            coord = np.array([[float(val) for val in line.split()] for line in f])
+        i = coord[:, 0].astype(int) - 1
+        j = coord[:, 1].astype(int) - 1
+        data = coord[:, 2].astype(dtype)
+        A = coo_matrix((data, (i, j)))
+    elif precision in {'complex8', 'complex16'}:
+        # Harwell-Boeing Exchange Format
+        if precision == 'complex8':
+            readhb_hdr, readhb = creadhb_hdr, creadhb
+        else:
+            readhb_hdr, readhb = zreadhb_hdr, zreadhb
+        nc, nz = readhb_hdr(path)
+        values = np.empty(nz, dtype=dtype)
+        colptr = np.empty(nc+1, dtype=np.int32)
+        rowind = np.empty(nz, dtype=np.int32)
+        res = readhb(path, values, colptr, rowind)
+        A = csc_matrix((values, rowind-1, colptr-1))
     return A
 
 
 def load_sigma(folder, precision="double", irl=False):
-    dtype = {"single": np.float32, "double": np.float64}[precision]
+    dtype = {
```

Comment:
This shows up a lot of places, should probably refactor so everything uses the same mapping

Reply:
Sounds like a good idea. Maybe `_dtype_map`?